src/main/java/Duke.java

```diff
@@ -1,10 +1,115 @@
+                    System.out.println("Wrong number of arguments for unmark");
+                    break;
+                }
+                try {
+                    int index = Integer.parseInt(wordList[1]) - 1;
+                    tasks.get(index).setDone(false);
+                    System.out.println("OK, I've marked this task as not done yet:");
+                    System.out.printf("[ ] %s\n", tasks.get(index).getDescription());
+                } catch (Exception exception) {
+                    System.out.println(exception);
+                }
+                break;
+            case "mark":
+                if (wordList.length != 2) {
+                    System.out.println("Wrong number of arguments for mark");
+                    break;
+                }
+                try {
+                    int index = Integer.parseInt(wordList[1]) - 1;
+                    tasks.get(index).setDone(true);
+                    System.out.println("Nice! I've marked this task as done:");
+                    System.out.printf("[X] %s\n", tasks.get(index).getDescription());
+
+                } catch (Exception exception) {
+                    System.out.println(exception);
+                }
+                break;
+            case "deadline":
+                try {
+                    line = line.substring(9);
+                    String description = line.split(" /by ")[0];
+                    String by = line.split(" /by ")[1];
+                    Deadline deadlineTask = new Deadline(description, by);
+                    addTask(tasks, deadlineTask);
+                    break;
+                } catch (Exception exception) {
+                    System.out.println(exception);
+                }
+                break;
+            case "todo":
+                try {
+                    String description = line.substring(5);
+                    Todo todoTask = new Todo(description);
+                    addTask(tasks, todoTask);
+                } catch (Exception exception) {
+                    System.out.println(exception);
+                }
+                break;
+            case "event":
+                try {
+                    line = line.substring(6);
+                    String description = line.split(" /from ")[0];
+                    String eventTime = line.split(" /from ")[1];
+                    String startTime = eventTime.split(" /to ")[0];
+                    String endTime = eventTime.split(" /to ")[1];
+                    Event eventTask = new Event(description, startTime, endTime);
+                    addTask(tasks, eventTask);
+                } catch (Exception exception) {
+                    System.out.println(exception);
+                }
+                break;
+            default:
+                System.out.println("command not found");
+            }
+            line = in.nextLine();
+        }
+        System.out.println("Bye. Hope to see you again soon!");
     }
```

Comment:
You could perhaps shorten this main method by using the extract method refactoring so that it would be easier for you to debug. Perhaps extract methods for each input command.